addons/payment/static/src/js/payment_processing.js

```diff
@@ -63,11 +65,7 @@ odoo.define('payment.processing', function (require) {
                     }
                 }
                 self.startPolling();
-            }).always(function(){
-                if ($.blockUI) {
-                    $.unblockUI();
-                }
-            }).fail(function(e) {
+             }).fail(function(e) {
```

Comment:
So if I understand correctly, we poll indefinitely unless the transaction succeeds at some point, in which case we open the redirect_url? Isn't it weird that we display an error and unblock the UI, but continue polling silently? Don't we have cases where the error is permanent, and polling is useless?
(it's probably outside the scope of this PR, sorry)

Reply:
Indeed a transaction can be stalled in the same state permanently, but it was impossible to know if it was due to a real error or a pg concurrency issue.
But now that we can detect "real" errors and pg concurrency one, we can stop the polling when needed.

It is something that can be done in master, @dbo-odoo *wink* *wink*